crates/viewer/re_space_view_spatial/src/visualizers/points2d.rs

```diff
@@ -211,53 +211,62 @@ impl VisualizerSystem for Points2DVisualizer {
         line_builder
             .radius_boost_in_ui_points_for_outlines(SIZE_BOOST_IN_POINTS_FOR_POINT_OUTLINES);
 
-        super::entity_iterator::process_archetype::<Self, Points2D, _>(
+        super::entity_iterator::process_archetype2::<Self, Points2D, _>(
             ctx,
             view_query,
             context_systems,
             |ctx, spatial_ctx, results| {
-                use re_space_view::RangeResultsExt as _;
+                use re_space_view::RangeResultsExt2 as _;
 
-                let resolver = ctx.recording().resolver();
-
-                let positions = match results.get_required_component_dense::<Position2D>(resolver) {
-                    Some(positions) => positions?,
-                    _ => return Ok(()),
+                let Some(all_position_chunks) = results.get_required_chunks(&Position2D::name())
+                else {
+                    return Ok(());
                 };
 
-                let num_positions = positions
-                    .range_indexed()
-                    .map(|(_, positions)| positions.len())
-                    .sum::<usize>();
+                let num_positions = all_position_chunks
+                    .iter()
+                    .flat_map(|chunk| chunk.iter_primitive_array::<2, f32>(&Position2D::name()))
+                    .map(|points| points.len())
+                    .sum();
+
                 if num_positions == 0 {
                     return Ok(());
                 }
 
                 point_builder.reserve(num_positions)?;
 
-                let colors = results.get_or_empty_dense(resolver)?;
-                let radii = results.get_or_empty_dense(resolver)?;
-                let labels = results.get_or_empty_dense(resolver)?;
-                let class_ids = results.get_or_empty_dense(resolver)?;
-                let keypoint_ids = results.get_or_empty_dense(resolver)?;
-
-                let data = range_zip_1x5(
-                    positions.range_indexed(),
-                    colors.range_indexed(),
-                    radii.range_indexed(),
-                    labels.range_indexed(),
-                    class_ids.range_indexed(),
-                    keypoint_ids.range_indexed(),
+                let timeline = ctx.query.timeline();
+                let all_positions_indexed = all_position_chunks.iter().flat_map(|chunk| {
+                    itertools::izip!(
+                        chunk.iter_component_indices(&timeline, &Position2D::name()),
+                        chunk.iter_primitive_array::<2, f32>(&Position2D::name())
+                    )
+                });
```

Comment:
it would be nice to have a similar utility like `iter_as` here to make this shorter

Reply:
:sob: I used to have them, but it seemed really over the top to maintain yet another set of helpers just for this part.

I'll try it once again and see how it feels.

Reply:
Brought them back! :bow: